src/Authentication/Authentication/Cmdlets/InvokeMgGraphRequest.cs

```diff
@@ -1003,11 +1003,8 @@ private string QualifyFilePath(string path)
         /// </summary>
         private void ResetGraphSessionEnvironment()
         {
-            var currentEnvironment = GraphSession.Instance.Environment;
-            if(currentEnvironment != null && !currentEnvironment.Equals(_originalEnvironment))
-            {
-                GraphSession.Instance.Environment = _originalEnvironment;
-            }
+            _originalEnvironment = GraphSession.Instance.Environment;
+            GraphSession.Instance.Environment = _originalEnvironment;
```

Comment:
The values will already be the same?

Reply:
``_originalEnvironment`` holds the previous state defined by the user. So to ensure subsequent requests by other cmdlets don't end up calling the default Global graph endpoint especially when connected to a national cloud, we assign the GraphSession.Instance.Environment to the value the `_originalEnvironment`` holds even after we are done using ``Invoke-MgGraphCommand``

Reply:
Thanks for sharing context. Still looks like we may not need one of the two lines since both variables will have the same value after line `1006`.